cluster-autoscaler/core/scaleup/resource/manager.go

```diff
@@ -180,6 +180,37 @@ func (m *Manager) ApplyLimits(ctx *context.AutoscalingContext, newCount int, res
 	return newCount, nil
 }
 
+// AllocateResources allocates the resources of required number of nodes from resources left and return true if resources are allocated successfully.
+func AllocateResources(left Limits, delta Delta, count int) bool {
+	if allocateResourcesWithDryRun(left, delta, count, true) {
+		return allocateResourcesWithDryRun(left, delta, count, false)
+	}
+	return false
+}
+
+func allocateResourcesWithDryRun(left Limits, delta Delta, count int, dryRun bool) bool {
+	for resourceType, resourceDelta := range delta {
+		if resourceDelta == 0 {
+			continue
+		}
+		resourceLeft, found := left[resourceType]
+		if !found {
+			continue
+		}
+		if resourceLeft == LimitUnknown {
```

Comment:
Could you add a unit test case for that?

Reply:
Added.